ocs_ci/ocs/ui/page_objects/alerting.py

```diff
@@ -0,0 +1,247 @@
+
+    def __repr__(self):
+        """
+        get base64 hash of the Raw runbook page content
+        """
+
+        return self.runbook_hash
+
+    def __eq__(self, other):
+        """
+        Compare two instances based on their hash values
+        """
+        if isinstance(other, Runbook):
+            return self.runbook_hash == other.runbook_hash
+        return False
+
+    def __ne__(self, other):
+        """
+        Compare two instances based on their hash values
+        """
+        return not self.__eq__(other)
+
+    def check_text_content(self, headers, *args):
+        """
+        Check if the text is valid
+
+        Returns:
+            bool: True if the text is valid, False otherwise
+        """
+        if hasattr(self, "text") and not self.text:
+            raise ValueError(
+                "Runbook text is empty, this object instantiated with hash value as Expected result for the test "
+                "and should not be checked"
+            )
+        # Check if the text contains all the headers
+        for chapter in headers:
+            if not re.search(f"## {headers}", self.text):
+                logger.error(f"Chapter '{chapter}' is missing.")
+                return False
+
+        # Check additional text arguments if provided
+        for arg in args:
+            if not re.search(arg, self.text):
+                logger.error(f"arg '{arg}' is missing int the text.")
+                return False
+
+        return True
+
+
+class Alerting(PageNavigator):
+    """
+    Alerting page that contains 3 tabs - Alerts, Silences and Alerting Rules
+    The default tab is Alerts
+    """
+
+    def __init__(self):
+        PageNavigator.__init__(self)
+
+    def nav_alerts(self):
+        """
+        Navigate to Alerts tab. Default tab when opening the Alerting page
+
+        Returns:
+            Alerts: Alerts page
+        """
+        logger.info("Navigate to Alerts tab")
+        self.do_click(self.validation_loc["alerts-tab-link"], enable_screenshot=True)
+        return Alerts()
+
+    def nav_silences(self):
+        """
+        Navigate to Silences tab
+
+        Returns:
+            Silences: Silences page
+        """
+        logger.info("Navigate to Silences tab")
+        self.do_click(self.alerting_loc["silences-tab-link"], enable_screenshot=True)
+        return Silences()
+
+    def nav_alerting_rules(self):
+        """
+        Navigate to Alerting Rules tab
+
+        Returns:
+            AlertingRules: Alerting Rules page
+        """
+        logger.info("Navigate to Alerting Rules tab")
+        self.do_click(
+            self.alerting_loc["alerting-rules-tab-link"], enable_screenshot=True
+        )
+        return AlertingRules()
+
+
+class Alerts(Alerting, SearchBar):
+    """
+    Alerts page object
+    """
+
+    def __init__(self):
+        Alerting.__init__(self)
+        SearchBar.__init__(self)
+
+    def search_alert(self, search_text: str):
```

Comment:
```suggestion
    def search_alert(self, search_text):
```

this is to maintain consistency in the repo

Reply:
not agree that using type hint is an issue.

removed because it is not worth for discussion